**Задача 1.** Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):

zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],

ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].

Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy

Полученные значения должны быть равны.

Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков, а затем с использованием функций из библиотек numpy и pandas.

In [1]:

import numpy as np
import pandas as pnd

zp = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])


Коэффициэнт корреляции Пирсона

In [2]:
np. corrcoef (zp, ks)

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

In [3]:
cov = np.mean(zp*ks) - np.mean (zp) * np.mean (ks) 
cov

9157.839999999997

Несмещенная ковариация

In [4]:
np.cov (zp, ks)

array([[ 3882.93333333, 10175.37777778],
       [10175.37777778, 33854.32222222]])

In [5]:
np.std (zp,ddof=1)

62.31318747531162

In [6]:
np.std (ks,ddof=1)

183.99544076477068

Коэффициэнт корреляции Пирсона

In [7]:
10175.37777778/(62.31318747531162*183.99544076477068)

0.88749009207411

Смещенная ковариация

In [8]:
np.cov (zp, ks, ddof = -1)

array([[ 3176.94545455,  8325.30909091],
       [ 8325.30909091, 27698.99090909]])

In [9]:
np.std (zp,ddof=-1)

56.364398821822405

In [10]:
np.std (ks,ddof=-1)

166.4301382234928

Коэффициэнт корреляции Пирсона

In [11]:
8325.30909091/(56.364398821822405*166.4301382234928)

0.8874900920740129

**Задача 2:** Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:

131, 125, 115, 122, 131, 115, 107, 99, 125, 111.

Известно, что в генеральной совокупности IQ распределен нормально.

Найдите доверительный интервал для математического ожидания с надежностью 0.95

In [12]:
from scipy import stats

st = np. array ([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])

In [13]:
x = np.mean(st)
x

118.1

In [14]:
D1 = np.var(st,ddof = 1)
D1

111.21111111111111

In [15]:
t = stats.t.ppf(0.975,9)
t

2.2621571627409915

In [16]:
x-t*np.sqrt(D1/10)

110.55608365158724

In [17]:
x+t*np.sqrt(D1/10)

125.64391634841274

Доверительный интервал: 110.56;125.64

**Задача 3:** Известно, что рост футболистов в сборной распределен нормально с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27, среднее выборочное составляет 174.2. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [18]:
x_1 = 174.2
t_1 = 1.96
D = 25
n = 27


In [19]:
x_1-t_1*np.sqrt(D/n)

172.31398912064722

In [20]:
x_1+t_1*np.sqrt(D/n)

176.08601087935276

In [21]:
mean_std = np.sqrt(25) / np.sqrt(27)
mean_std

0.9622504486493763

In [22]:
from statsmodels.stats.weightstats import _tconfint_generic as t_stat

t_stat(174.2, mean_std,27 - 1, 0.05, 'two-sided')

(172.2220658754539, 176.17793412454608)

In [25]:
mean_std_1 = np.sqrt(D1) / np.sqrt(10)
mean_std_1

3.3348329959851224

In [26]:
t_stat(x, mean_std_1,10 - 1, 0.05, 'two-sided')

(110.55608365158724, 125.64391634841274)